CRM/Utils/Check.php

```diff
@@ -209,7 +212,7 @@ public static function checkAll($max = FALSE) {
 
     foreach ($messages as $key => $message) {
       $hush = self::checkHushSnooze($message);
-      $message->setVisible(!$hush);
+      $messages[$key]->setVisible(!$hush);
```

Comment:
So I first thought this was a problem, then you pointed out that the API was working fine, which got me thinking this was redundant, then I tried removing it, and everything broke.  

I _think_ the reason it had been working was that even though $message wasn't passed by reference, because the setVisible() method goes and sets the property, it must be clear somehow that this is the actual $messages[$key] instance rather than a copy of $message that'll get left behind as you're foreaching.  Obviously it was doing the job, but I don't trust it, and besides, $messages[$key] should make it explicit and work no worse.

The real fix for CRM-17575 is what I added at line 84.